doc/builders/images/dockertools.section.md

```diff
@@ -394,3 +394,114 @@ buildImage {
   };
 }
 ```
+
+## buildNixShellImage {#ssec-pkgs-dockerTools-buildNixShellImage}
+
+Create a Docker image that sets up the `nix-shell` environment of a derivation. With Docker, this environment is similarly sandboxed as if the derivation were built with `nix-build`, with a major difference being that access to the internet is allowed. It additionally also behaves like an interactive `nix-shell`, running things like `shellHook` and setting an interactive prompt. If the derivation is fully buildable (i.e. `nix-build` can be used on it), running `buildDerivation` inside such a Docker image will build the derivation, with all its outputs being available in the correct `/nix/store` paths, pointed to by the respective environment variables like `$out`, etc.
+
+::: {.warning}
+The behavior doesn't match nix-shell exactly and this function may not work correctly for e.g. fixed-output derivations, CA derivations, impure derivations and other special types of derivations.
+:::
+
+`drv`
+
+: The derivation whose environment this Docker image should be based on.
+
+    Adding additional packages to the Docker image is possible by e.g. extending the list of `nativeBuildInputs` of this derivation like
+
+    ```nix
+    buildNixShellImage {
+      drv = someDrv.overrideAttrs (old: {
+        nativeBuildInputs = old.nativeBuildInputs or [] ++ [
+          somethingExtra
+        ];
+      });
+      # ...
+    }
+    ```
+
+    Similarly, you can extend the image initialization script by extending `shellHook`
+
+`name` _optional_
+
+: The name of the resulting image.
+
+    *Default:* the `drv`'s name + `"-env"`
+
+`tag` _optional_
+
+: Tag of the generated image.
+
+    *Default:* the resulting image derivation output path's hash
```

Comment:
This is a handful... Also we seem to be using output path and store path interchangibly. Are they synonymous? If yes, we should use exactly one of them everywhere.
```suggestion
    *Default:* the store path hash of the resulting image's derivation
```

Reply:
This was also discussed in https://github.com/NixOS/nixpkgs/pull/172736#discussion_r873874868. Notably it's not the store path hash of the resulting image's derivation, but rather some internal `config.json` file.